appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecService.java

```diff
@@ -43,28 +61,151 @@ public static AppSecService getInstance() {
         return AppSecService.InstanceHolder.instance;
     }
 
+    private final List<AppSecScanEventListener> scanEventListeners = new ArrayList<>();
+
+    private final OpenSourceVulnerabilityService osvService;
+
+    private final VulnerabilityStore vulnerabilityStore;
+
+    private final BillOfMaterialsStore bomStore;
+
+    private final AppSecDTOProvider dtoProvider;
+
     private AppSecConfiguration configuration;
 
     private AppSecData data;
 
+    private Clock clock = Clock.systemUTC();
+
+    private ScheduledFuture<?> scheduledScan;
+
     /**
      * Creates a new instance of the service.
      *
      * @param configuration
      *            the configuration bean
      */
     private AppSecService(AppSecConfiguration configuration) {
+        bomStore = new BillOfMaterialsStore();
+        osvService = new OpenSourceVulnerabilityService();
+        vulnerabilityStore = new VulnerabilityStore(osvService, bomStore);
+        dtoProvider = new AppSecDTOProvider(vulnerabilityStore, bomStore);
         this.configuration = configuration;
     }
 
+    /**
+     * Initializes the service reading the SBOM file.
+     */
+    public void init() {
+        cancelScheduledScan();
+        Path bomFilePath = configuration.getBomFilePath();
+        try {
+            bomStore.readBomFile(bomFilePath);
+        } catch (ParseException e) {
+            throw new AppSecException(
+                    "Cannot parse the SBOM file: " + bomFilePath.toString(), e);
+        }
+        readOrCreateDataFile();
+    }
+
+    /**
+     * Schedules automatic scan for vulnerabilities at a fixed rate set to the
+     * value configured with
+     * {@link AppSecConfiguration#setAutoScanInterval(java.time.Duration)}.
+     */
+    public void scheduleAutomaticScan() {
```

Comment:
Should we call the `checkForInitialization()` function in this public API too?

Reply:
We should, added now.